### ISSUE TITLE
Forced subtitles are loading instead of default one

### ISSUE THREAD

**Issue by locobastos** [clsid2/mpc-hc#1551]
## Environment
```
MPC-HC (64-bit)
---------------

Build information:
    Version:            1.9.19 (b64326a16)
    Compiler:           MSVC v19.29.30139
    Build date:         Jan 30 2022

LAV Filters:
    LAV Splitter:       0.76.0.2
    LAV Video:          0.76.0.2
    LAV Audio:          0.76.0.2
    FFmpeg compiler:    MinGW-w64 GCC 10.2.0

Operating system:
    Name:               Windows 10 (Build 21H2)
    Version:            10.0.19044 (64-bit)

Hardware:
    CPU:                Intel(R) Core(TM) i5-2500K CPU @ 3.30GHz
    GPU1:               NVIDIA GeForce GTX 1060 3GB (driver version: 30.0.15.1123)
    GPU2:               NVIDIA GeForce GTX 1060 3GB (driver version: 30.0.15.1123)
    GPU3:               NVIDIA GeForce GTX 1060 3GB (driver version: 30.0.15.1123)
```

## Steps to reproduce
Open an MKV file with at least 2 subtitles tracks:
```
Text #1
ID                                       : 4
Format                                   : UTF-8
Codec ID                                 : S_TEXT/UTF8
Codec ID/Info                            : UTF-8 Plain Text
Duration                                 : 22 min 46 s
Bit rate                                 : 45 b/s
Count of elements                        : 257
Stream size                              : 7.61 KiB (0%)
Title                                    : English (FULL)
Language                                 : English
Default                                  : Yes
Forced                                   : No

Text #2
ID                                       : 5
Format                                   : UTF-8
Codec ID                                 : S_TEXT/UTF8
Codec ID/Info                            : UTF-8 Plain Text
Duration                                 : 2 min 34 s
Bit rate                                 : 5 b/s
Count of elements                        : 3
Stream size                              : 97.0 Bytes (0%)
Title                                    : English (FORCED)
Language                                 : English
Default                                  : No
Forced                                   : Yes
```

## Actual result
The forced track is loaded

## Expected result
The default track is loaded

## Reporter note
If I understand well the Matroska mechanics, tags/flags are used by some players (software or hardware) to fill the needs of the watchers. Of course, the player's parameters must be set correctly.

"Hearing impaired" flag means the sounds are described, "Forced display" means that subtitles are shown only during foreign parts of the video, "Default track" means this is the track we want to show by default, etc.

I know most people will use original audio language + forced subtitles by default but this is not logic.
If you want forced track by default, add "Default track" flag to the forced subtitle track.
But if the "default track" flag is set on another subtitle track (let's say the "English (FULL)" track because it also has the "Hearing Impaired" flag and content), MPC-HC should load the default one by default and not the forced track...

I hope I have described correctly my problem.
Thank you for your help.
Best Regards.

PS: No, I really don't want to switch to VLC. I prefer to change the subtitles track every video than switching to it.

**Comment by xLn2** [comment]
Did you try 

1. Options > Subtitles > Misc > Prefer forced and/or default subtitles tracks
2. Options > Subtitles > Misc > Prefer external subtitles over embedded subtitles
3. Options > Internal Filters > Splitter > check default and advanced selection mode in order to your needs. ( https://1f0.de/lav-splitter/lav-splitter-stream-selection/)

**Comment by locobastos** [comment]
By default, my options are:

- Options > Subtitles > Misc > Prefer forced and/or default subtitles tracks --> ticked
- Options > Subtitles > Misc > Prefer external subtitles over embedded subtitles --> ticked
- Options > Internal Filters > Splitter --> Default

I had no effect when I played with the two checkboxes, but I succeed in setting correct parameters in the Internal Filters > Splitter > Advanced mode:

    *:eng;eng:eng|f

I guess my need is filled by following your comment.
Thank you very much.
I let you close this issue if the main developers do not consider this as an issue.

**Comment by Alp1821** [comment]
This is not bug. Your 1st subtitle track is not flagged as default one.
So mpc is preferring forced one.

**Comment by Alp1821** [comment]
Download Mkvtoolnix. Open the file. Flag your preferred subtitle as default. Then remux it.
Mpc will load default one.

**Comment by clsid2** [comment]
The default track should get selected with default settings. If you believe there is a bug in the splitter, then I need a link to a sample file. I am too lazy to create one.

**Comment by locobastos** [comment]
To be sure that my parameters are not wrong:
1. I have downloaded [MPC-HC.1.9.19.x64.zip](https://github.com/clsid2/mpc-hc/releases/download/1.9.19/MPC-HC.1.9.19.x64.zip) (I prefer to use portable software, but I can install it if it can help you)
2. I have extracted it on an empty folder
3. I have reset all the parameters: View > Options > Miscellaneous > Reset > Yes
4. Download and play this sample: [GOT.S02E10.SAMPLE.mkv](https://anonfiles.com/x6OcZeH0x1/GOT.S02E10.SAMPLE_mkv)

```
General
Format                                   : Matroska
Format version                           : Version 4
File size                                : 27.0 MiB
Duration                                 : 1 min 35 s
Overall bit rate                         : 2 378 kb/s
Movie name                               : Game of Thrones S02E10: Valar Morghulis
Encoded date                             : UTC 2022-02-14 16:44:42
Writing application                      : mkvmerge v65.0.0 ('Too Much') 64-bit
Writing library                          : libebml v1.4.2 + libmatroska v1.6.4

Video
ID                                       : 1
Format                                   : AVC
Format/Info                              : Advanced Video Codec
Format profile                           : High@L4.1
Format settings                          : CABAC / 4 Ref Frames
Format settings, CABAC                   : Yes
Format settings, Reference frames        : 4 frames
Codec ID                                 : V_MPEG4/ISO/AVC
Duration                                 : 1 min 31 s
Bit rate                                 : 2 074 kb/s
Width                                    : 1 920 pixels
Height                                   : 1 080 pixels
Display aspect ratio                     : 16:9
Frame rate mode                          : Constant
Frame rate                               : 23.976 (23976/1000) FPS
Original frame rate                      : 23.976 (24000/1001) FPS
Color space                              : YUV
Chroma subsampling                       : 4:2:0
Bit depth                                : 8 bits
Scan type                                : Progressive
Bits/(Pixel*Frame)                       : 0.042
Stream size                              : 22.6 MiB (84%)
Title                                    : Valar Morghulis
Language                                 : English
Default                                  : Yes
Forced                                   : No
Color range                              : Limited
Color primaries                          : BT.709
Transfer characteristics                 : BT.709
Matrix coefficients                      : BT.709

Audio
ID                                       : 2
Format                                   : AC-3
Format/Info                              : Audio Coding 3
Commercial name                          : Dolby Digital
Codec ID                                 : A_AC3
Duration                                 : 1 min 35 s
Bit rate mode                            : Constant
Bit rate                                 : 384 kb/s
Channel(s)                               : 2 channels
Channel layout                           : L R
Sampling rate                            : 48.0 kHz
Frame rate                               : 31.250 FPS (1536 SPF)
Bit depth                                : 32 bits
Compression mode                         : Lossy
Delay relative to video                  : -3 s 737 ms
Stream size                              : 4.35 MiB (16%)
Title                                    : English
Language                                 : English
Service kind                             : Complete Main
Default                                  : No
Forced                                   : No

Text #1
ID                                       : 3
Format                                   : UTF-8
Codec ID                                 : S_TEXT/UTF8
Codec ID/Info                            : UTF-8 Plain Text
Duration                                 : 1 min 27 s
Bit rate                                 : 58 b/s
Count of elements                        : 16
Stream size                              : 641 Bytes (0%)
Title                                    : English (FULL)
Language                                 : English
Default                                  : Yes
Forced                                   : No

Text #2
ID                                       : 4
Format                                   : UTF-8
Codec ID                                 : S_TEXT/UTF8
Codec ID/Info                            : UTF-8 Plain Text
Duration                                 : 1 min 27 s
Bit rate                                 : 77 b/s
Count of elements                        : 16
Stream size                              : 849 Bytes (0%)
Title                                    : English (FORCED)
Language                                 : English
Default                                  : No
Forced                                   : Yes
```

If the sample is not what you expected, let me know.

**Comment by clsid2** [comment]
@nevcairiel
Seems to be a regression in LAV Splitter.
I tested with GraphStudioNext and it selected the second forced track instead of first default track. Settings all at defaults.

[Sample file](https://anonfiles.com/x6OcZeH0x1/GOT.S02E10.SAMPLE_mkv)

**Comment by locobastos** [comment]
FYI, this behaviour affects the original MPC-HC 1.7.13 (July 16, 2017)
If it is really a regression, it may be very old.

**Comment by Nevcairiel** [comment]
If you do not specify a preferred subtitle language in LAV, then full subtitles are not automatically enabled. Only if you actually specify a language will it start using full subtitles in that language.

Seems to be behaving as described for me. With no language, it loads forced, with a language it loads Default.

I don't think this behavior changed, as I believe this is how it was always meant to work.

TLDR; if you want full subtitles, tell it the language(s) you like!

(Edit: Looking at the logic, its a bit weird, the behavior in this kind of file has certainly been the same ever since it was made, but its slightly confused logic. So its not a regression, but the result is not ideal. Might change it for the next major update)

**Comment by clsid2** [comment]
Fixed/changed in LAV.
Will be in next MPC-HC build.

Thanks.
